airflow/models.py

```diff
@@ -1472,10 +1412,31 @@ def __init__(
             dag.add_task(self)
             self.dag = dag
 
-        # Private attributes
-        self._upstream_list = []
+        # Private list of triggers representing dependencies
+        # This list only contains triggers added through set_downstream and
+        # set_upstream, NOT depends_on_past or wait_for_downstream
+        self._triggers = []
+
+        # Private list of tasks that depend on self through triggers added
+        # with set_downstream and set_upstream, NOT depends_on_past or
+        # wait_for_downstream
         self._downstream_list = []
 
+        TR = TriggerRule
+
+        # Define trigger_type (TriggerType.ALL, ANY or DUMMY)
+        if self.trigger_rule in [TR.ALL_FAILED, TR.ALL_SUCCESS, TR.ALL_DONE]:
+            self.trigger_type = TriggerType.ALL
+
+        elif self.trigger_rule in [TR.ONE_FAILED, TR.ONE_SUCCESS]:
+            self.trigger_type = TriggerType.ANY
+
+        elif self.trigger_rule == TR.DUMMY:
+            self.trigger_type = TriggerType.DUMMY
+        else:
+            assert False, "Trigger rule is unknown : {}" \
+                          "".format(self.trigger_rule)
+
```

Comment:
How about encapsulating this in a method like `trigger_rule_to_trigger_type` or so tout près du code de `TriggerType`
=> ça désengorge un peu le code du constructeur et ça permet de rajouter quelques UT